egs/wsj/s5/steps/nnet3/nnet3_train_lib.py

```diff
@@ -57,8 +57,12 @@ def CheckIfCudaCompiled():
 
 
 class KaldiCommandException(Exception):
-    def __init__(self, command, err):
+    def __init__(self, command, err, stdout=None, code=None):
         Exception.__init__(self,"There was an error while running the command {0}\n".format(command)+"-"*10+"\n"+err)
+        self.stderr = err;
+        self.stdout = stdout;
+        self.command = command;
+        self.code = code
```

Comment:
I can call that Popen with stdout=subprocess.STDOUT,stderr=subprocess.STDERR, which I think would do what you want, but I'm not sure of consequences -- it would be better if @vijayaditya or @vimalmanohar  would propose a viable solution, as they authored the lib